map/src/infoblock/components/wpt/WptDetails.jsx

```diff
@@ -0,0 +1,350 @@
+            hidden: currentWpt.hidden,
+            latlon: { lat: currentWpt.lat, lon: currentWpt.lon },
+            marker: currentWpt.marker,
+            background: prepareBackground(currentWpt.background),
+            color: prepareColor(currentWpt.color),
+            icon: prepareIcon(currentWpt.icon),
+            category: currentWpt.category,
+            address: currentWpt.address,
+            time: parseInt(currentWpt.ext?.time) !== 0 ? currentWpt.ext.time : null,
+            tags: WptTagsProvider.getWptTags(currentWpt, type),
+        };
+    }
+
+    useEffect(() => {
+        if (newWpt !== null) {
+            setWpt(newWpt);
+        }
+    }, [newWpt]);
+
+    function getWptType(wpt) {
+        return {
+            isPoi: ctx.currentObjectType === OBJECT_TYPE_POI && wpt?.poi,
+            isWpt: isTrack(ctx) && (wpt?.trackWpt || wpt?.trackWptItem),
+            isFav: ctx.currentObjectType === OBJECT_TYPE_FAVORITE && wpt?.markerCurrent,
+        };
+    }
+
+    function closeDetails() {
+        if (wpt.type?.isPoi) {
+            closeHeader({ ctx });
+        } else if (wpt.type?.isWpt) {
+            isDetails ? setOpenWptTab(true) : closeHeader({ ctx });
+        } else if (wpt.type?.isFav) {
+            isDetails ? closeOnlyFavDetails() : closeHeader({ ctx });
+        }
+        ctx.setSelectedWpt(null);
+    }
+
+    function closeOnlyFavDetails() {
+        setShowInfoBlock(false);
+        ctx.setCurrentObjectType(null);
+        ctx.setSelectedGpxFile({});
+    }
+
+    function getId() {
+        if (wpt.type?.isFav) {
+            return 'se-fav-item-info-' + wpt.name;
+        }
+        if (wpt.type?.isPoi) {
+            return 'se-poi-infoblock-' + wpt.name;
+        }
+        return null;
+    }
+
+    function formatTime(time) {
+        const locale = locales[i18n.language] || locales.enUS;
+        return format(time, 'MMM dd, yyyy – HH:mm', { locale: locale }).replace(',', '');
+    }
+
+    const Header = () => {
+        return (
+            <AppBar position="static" className={headerStyles.appbar}>
+                <Toolbar className={headerStyles.toolbar}>
+                    <IconButton
+                        variant="contained"
+                        type="button"
+                        className={styles.closeIcon}
+                        onClick={() => closeDetails()}
+                    >
+                        {isDetails ? <BackIcon /> : <CloseIcon />}
+                    </IconButton>
+                </Toolbar>
+            </AppBar>
+        );
+    };
+
+    const WptIcon = () => {
+        return (
+            <div
+                style={{ display: 'flex' }}
+                dangerouslySetInnerHTML={{
+                    __html:
+                        changeIconSizeWpt(
+                            removeShadowFromIconWpt(
+                                MarkerOptions.getWptIcon(wpt, wpt?.color, wpt?.background, wpt?.icon).options.html
```

Comment:
getWptIcon() has 5 parameters but only 4 are used everywhere, so please recheck and possibly remove unused `folder` parameter